tasks/build/download_node_builds.js

```diff
@@ -1,84 +1,134 @@
-module.exports = function (grunt) {
-  let { map, fromNode } = require('bluebird');
-  let { resolve } = require('path');
-  let { pluck } = require('lodash');
-  let { createWriteStream } = require('fs');
-  let { createGunzip } = require('zlib');
-  let { Extract } = require('tar');
-  let { rename } = require('fs');
-  let wreck = require('wreck');
-
-  let platforms = grunt.config.get('platforms');
-  let activeDownloads = [];
-
-  let start = async (platform) => {
-    let finalDir = platform.nodeDir;
-    let downloadDir = `${finalDir}.temp`;
-
-    if (grunt.file.isDir(platform.nodeDir)) {
-      grunt.log.ok(`${platform.name} exists`);
-      return;
-    }
+import { map, fromNode, promisify } from 'bluebird';
+import { resolve, basename, dirname } from 'path';
+import { createReadStream, createWriteStream, writeFile } from 'fs';
+import { createGunzip } from 'zlib';
+import { Extract } from 'tar';
+import { rename } from 'fs';
+import checkHash  from 'check-hash';
+import wreck from 'wreck';
 
-    let resp = await fromNode(cb => {
-      let req = wreck.request('GET', platform.nodeUrl, null, function (err, resp) {
-        if (err) {
-          return cb(err);
-        }
+const wreckGet = promisify(wreck.get, wreck);
+const checkHashFromFile = promisify(checkHash.fromFile);
+const renameAsync = promisify(rename);
+const writeFileAsync = promisify(writeFile);
+
+module.exports = function (grunt) {
+  const platforms = grunt.config.get('platforms');
+  const downloadLimit = 3;
 
-        if (resp.statusCode !== 200) {
-          return cb(new Error(`${platform.nodeUrl} failed with a ${resp.statusCode} response`));
-        }
+  let shaSums = {};
+  const getShaSums = () => {
+    const nodeVersion = grunt.config.get('nodeVersion');
+    const shaSumsUri = `https://nodejs.org/dist/v${nodeVersion}/SHASUMS256.txt`;
 
-        return cb(null, resp);
+    return wreckGet(shaSumsUri).then(([resp, payload]) => {
+      if (resp.statusCode !== 200) {
+        throw new Error(`${shaSumsUri} failed with a ${resp.statusCode} response`);
+      }
+      payload.toString('utf8fix/7136 ').split('\n').forEach(line => {
+        const [sha, platform] = line.split('  ');
+        shaSums[platform] = sha;
       });
```

Comment:
Yeah, most chains should be split out on separate lines. I don't agree with some of our style guide rules myself, but I believe consistency across the team trumps any single rule.

https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#chaining-operations